### ISSUE TITLE
Inconsistent Behavior Inside the Custom Open File Dialog of Remote Extensions when Navigating with a Screen Reader 

### ISSUE THREAD

**Issue by dchemishanov** [microsoft/vscode#109722]
- VSCode Version: latest september 2020
- OS Version: Windows 10

Steps to Reproduce:

1. Open a remote session e.g. Remote WSL extension
2. Open a folder inside the remote environment. The folder should contain multiple subfolders and files. the names of Part of the folders and files should begin with the same substring, e.g. "device_".
3. Ctrl+o to open the open file dialog.
4. Type the first few letters of the common substring, e.g. "dev".
5. Arrow down the results offered.

Expected behavior:
1. If the open file dialog is based on autocomplete, the offered options must all start with "dev". Instead, the offered results are the same as if the cursor is moved to the first folder starting with "dev". It results in the files starting with "dev" being hard to find unless the full name of the file is typed.
2. If the dialog is based on first-letter-navigation, typing "dev", should move the cursor to the first item whose name starts with the given letter. Typing the next letter in the sequence should behave the same, but remembering the previous position of the cursor and moving from there.

Does this issue occur when all extensions are disabled?: no.

**Comment by dchemishanov** [comment]
@isidorn I'm tagging @alexr00 as requested.

**Comment by isidorn** [comment]
That is due to the fact that we use a custom open dialog when connected to a Remote. We should make it more screen reader friendly, since users are expecting it to behave like a native dialog.
Assigning first to @alexr00

**Comment by alexr00** [comment]
> If the dialog is based on first-letter-navigation

(2) above is how the simple file dialog works. However, the change requested would be a significant change for existing users of the file dialog, and I don't want to break their muscle memory. As a middle ground, I have tried to make a small improvement for when VS Code is optimized for screen readers (`editor.accessibilitySupport` set to `on` or sometimes to `auto`).

**Comment by alexr00** [comment]
To verify, 

Set `editor.accessibilitySupport` to `off`.
Set `files.simpleDialog.enable` to `true`.
Run the Open File command.
Type a letter that matches something in the folder that the simple file dialog opened to. 
Arrow down.
Verify that the the input box has ad the file item that you arrowed down to appended and that the last segment of the path is selected.
Set `editor.accessibilitySupport` to `on`.
Run the Open File command.
Type a letter that matches something in the folder that the simple file dialog opened to. 
Arrow down.
Verify that the contents of the input box haven't changed, and that if you start typing more it moves the list selection appropriately.
